homeassistant/components/mazda/binary_sensor.py

```diff
@@ -0,0 +1,184 @@
+"""Platform for Mazda binary sensor integration."""
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import MazdaEntity
+from .const import DATA_CLIENT, DATA_COORDINATOR, DOMAIN
+
+
+@dataclass
+class MazdaBinarySensorRequiredKeysMixin:
+    """Mixin for required keys."""
+
+    # Suffix to be appended to the vehicle name to obtain the binary sensor name
+    name_suffix: str
+
+    # Function to determine the value for this binary sensor, given the coordinator data
+    value: Callable[[dict], bool]
+
+
+@dataclass
+class MazdaBinarySensorEntityDescription(
+    BinarySensorEntityDescription, MazdaBinarySensorRequiredKeysMixin
+):
+    """Describes a Mazda binary sensor entity."""
+
+    # Function to determine whether the vehicle supports this binary sensor, given the coordinator data
+    is_supported: Callable[[dict], bool] = lambda data: True
+
+    # Function to return extra state attributes, given the coordinator data
+    state_attributes: Callable[[dict], dict] | None = None
```

Comment:
Same as above, please make `dict` more specific

Reply:
Changed to `dict[str, Any]`.